app/questionnaire/path_finder.py

```diff
@@ -43,22 +44,57 @@ def routing_path(
         )
 
         if section:
+            block_ids_dependent_on_when_rules = (
+                self._get_block_ids_dependent_on_when_rules(section)
+            )
+            blocks = self._get_not_skipped_blocks_in_section(
+                current_location,
+                routing_path_block_ids,
+                section,
+                block_ids_dependent_on_when_rules,
+            )
+
+            if blocks:
+                routing_path_block_ids = self._build_routing_path_block_ids(
+                    blocks, current_location, block_ids_dependent_on_when_rules
+                )
+
+        return RoutingPath(routing_path_block_ids, section_id, list_item_id, list_name)
+
+    def _get_block_ids_dependent_on_when_rules(
+        self, section: ImmutableDict
```

Comment:
From [our type hints guide](https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/doc/python-type-hinting.md#generic-types): specify at least the top-level type parameters for all generic types?

Reply:
That is the type coming back from schema, what would you suggest?

Reply:
Comes from:
```python
self.schema.get_section(section_id)
```
and returns things like:
`ImmutableDict({'id': 'section', 'groups': (ImmutableDict({'id': 'group', 'title': 'List', 'blocks': (ImmutableDict({'type': 'Question', 'id': 'address-block-mandatory', 'question': ImmutableDict({'type': 'General', 'id': 'address-question-mandatory', 'title': 'What is your current address?', 'answers': (ImmutableDict({'id': 'address-mandatory', 'mandatory': True, 'type': 'Address'}),)})}), ImmutableDict({'type': 'Question', 'id': 'address-block-optional', 'question': ImmutableDict({'type': 'General', 'id': 'address-question-optional', 'title': 'What was your address one year ago?', 'answers': (ImmutableDict({'id': 'address-optional', 'mandatory': False, 'type': 'Address'}),)})}), ImmutableDict({'type': 'Question', 'id': 'address-confirmation', 'question': ImmutableDict({'type': 'General', 'id': 'address-confirmation-question', 'title': ImmutableDict({'placeholders': (ImmutableDict({'placeholder': 'household_address_line1', 'value': ImmutableDict({'identifier': 'address-mandatory', 'source': 'answers', 'selector': 'line1'})}),), 'text': 'Please confirm the first line of your address is {household_address_line1}'}), 'answers': (ImmutableDict({'type': 'Radio', 'id': 'address-confirmation-answer', 'mandatory': False, 'options': (ImmutableDict({'label': 'Yes', 'value': 'Yes'}), ImmutableDict({'label': 'No', 'value': 'No'}))}),)})}))}),)})`

so `ImmutableDict[str, Any]`?